hibernate-core/src/test/java/org/hibernate/orm/test/mapping/basic/CurrencyMappingTests.java

```diff
@@ -46,17 +64,146 @@ public void verifyMappings(SessionFactoryScope scope) {
 		);
 
 		scope.inTransaction(
-				(session) -> session.find(EntityWithCurrency.class, 1)
+			(session) -> session.find(EntityWithCurrency.class, 1)
 		);
 	}
 
+	@Test
+	public void basicAssertions(final SessionFactoryScope scope) {
+		final SessionFactoryImplementor sessionFactory = scope.getSessionFactory();
+		final JdbcTypeRegistry jdbcTypeRegistry = sessionFactory.getTypeConfiguration().getJdbcTypeRegistry();
+		final EntityPersister entityDescriptor = sessionFactory.getMappingMetamodel().getEntityDescriptor(
+				EntityWithCurrency.class );
+		{
+			final BasicAttributeMapping currencyAttribute = (BasicAttributeMapping) entityDescriptor.findAttributeMapping(
+					"currency");
+			Assertions.assertThat(currencyAttribute.getJdbcMapping().getJdbcType())
+					.isEqualTo(jdbcTypeRegistry.getDescriptor(Types.VARCHAR));
+			Assertions.assertThat(currencyAttribute.getJdbcMapping().getJavaTypeDescriptor().getJavaTypeClass()).isEqualTo(
+					Currency.class );
+		}
+		{
+			final PluralAttributeMapping currenciesAttribute = (PluralAttributeMapping) entityDescriptor.findAttributeMapping(
+					"currencies" );
+			final BasicValuedCollectionPart elementDescriptor = (BasicValuedCollectionPart) currenciesAttribute.getElementDescriptor();
+			Assertions.assertThat( elementDescriptor.getJdbcMapping().getJdbcType() )
+					.isEqualTo( jdbcTypeRegistry.getDescriptor( Types.VARCHAR ) );
+			Assertions.assertThat( elementDescriptor.getJdbcMapping().getJavaTypeDescriptor().getJavaTypeClass() ).isEqualTo(
+					Currency.class );
+		}
+	}
+
+	@Test
+	public void testCurrencyJavaTypeWrapNoOptionsPass(final SessionFactoryScope scope) {
+		final CurrencyJavaType currencyJavaType = new CurrencyJavaType();
+		{
+			final Currency usingNull = currencyJavaType.wrap(null, null);
+			Assertions.assertThat(usingNull).isNull();
+		}
+		{
+			final Currency usingString = currencyJavaType.wrap("CHF", null);
+			Assertions.assertThat(usingString).isNotNull();
+		}
+		{
+			final Currency usingCurrency = currencyJavaType.wrap(Currency.getInstance("CHF"), null);
+			Assertions.assertThat(usingCurrency).isNotNull();
+		}
+	}
+
+	@Test
+	public void testCurrencyJavaTypeWrapNoOptionsFail(final SessionFactoryScope scope) {
+		final CurrencyJavaType currencyJavaType = new CurrencyJavaType();
+		{
+			final String usingEmptyString = "";
+			Assertions.assertThatThrownBy(() ->
+				currencyJavaType.wrap(usingEmptyString, null)
+			).isInstanceOf(IllegalArgumentException.class);
+		}
+		{
+			final Integer usingInteger = Integer.valueOf(269);
+			Assertions.assertThatThrownBy(() ->
+				currencyJavaType.wrap(usingInteger, null)
+			).isInstanceOf(HibernateException.class);
+		}
+		{
+			final CurrencyJavaType usingSelf = new CurrencyJavaType();
+			Assertions.assertThatThrownBy(() ->
+				currencyJavaType.wrap(usingSelf, null)
+			).isInstanceOf(HibernateException.class);
+		}
+	}
+
+	@Test
+	public void testCurrencyJavaTypeWrapWithOptionsPass(final SessionFactoryScope scope) {
```

Comment:
I think you can simplify your tests: since the `WrapperOptions` parameter is never used neither for `YearJavaType` nor for `CurrencyJavaType` when wrapping/unwrapping you can just test once with `null` and avoid duplicating test methods.